### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by metaMMA**
- Log to `.log` file in root directory instead of syslog
- Add ability to display log in browser from config page
- Add ability to change log levels from config page

### CODE REVIEW

**Comment by metaMMA** [comment]
Still need to amend update.py
EDIT: done

File: weather.py [hunk 1]
```diff
@@ -197,19 +213,17 @@ def __init__(self):
...
-                    print('Driver: {0} failed.'.format(driver))
-                    syslog.syslog('Driver: {0} failed.'.format(driver))
+                    self.log.debug("Driver: {driver} failed.")
                     continue
                 found = True
                 break
 
             if not found:
-                raise Exception('No suitable video driver found!')
+                self.log.exception("No suitable video driver found!")
```
Comment (genebean) [inline weather.py:223]: This is a fatal exception that should cause the program to exit. Does this logging setup account for that?

**Review by genebean — CHANGES_REQUESTED** [review]
This isn't working for me @metaMMA 

```
$ export DISPLAY=:0
$ /usr/bin/python3 /home/pi/PiWeatherRock/weather.py
pygame 1.9.6
Hello from the pygame community. https://www.pygame.org/contribute.html
Traceback (most recent call last):
  File "/home/pi/PiWeatherRock/weather.py", line 806, in <module>
    MY_DISP = MyDisplay()
  File "/home/pi/PiWeatherRock/weather.py", line 192, in __init__
    self.log = get_logger()
  File "/home/pi/PiWeatherRock/weather.py", line 168, in get_logger
    handler = logging.handlers.RotatingFileHandler(
AttributeError: module 'logging' has no attribute 'handlers'
```

File: weather.py (hunk 1 above)
Comment (metaMMA) [reply weather.py:223]: No, it doesn't. As it stands now, the service file will continue to restart the script every 5 seconds either way. I think it should be fairly easy to handle this exception in the `start=stop` PR, though. We should be able to simply log the error and set `running` to `False`

**Comment by metaMMA** [comment]
> This isn't working for me @metaMMA
> 
> ```
> $ export DISPLAY=:0
> $ /usr/bin/python3 /home/pi/PiWeatherRock/weather.py
> pygame 1.9.6
> Hello from the pygame community. https://www.pygame.org/contribute.html
> Traceback (most recent call last):
>   File "/home/pi/PiWeatherRock/weather.py", line 806, in <module>
>     MY_DISP = MyDisplay()
>   File "/home/pi/PiWeatherRock/weather.py", line 192, in __init__
>     self.log = get_logger()
>   File "/home/pi/PiWeatherRock/weather.py", line 168, in get_logger
>     handler = logging.handlers.RotatingFileHandler(
> AttributeError: module 'logging' has no attribute 'handlers'
> ```

Copy and paste error on my part. I missed the last standard import: `import logging.handlers`.

**Review by genebean — APPROVED** [review]

**Pull request merged**